ibis/backends/sql/dialects.py

```diff
@@ -108,13 +109,86 @@ def _interval_with_precision(self, e):
     return f"INTERVAL {formatted_arg} {unit}"
 
 
+def _explode_to_unnest():
+    """Convert explode into unnest.
+
+    NOTE: Flink doesn't support UNNEST WITH ORDINALITY or UNNEST WITH OFFSET.
```

Comment:
Is this the reason you copied the implementation over? If so, I don't think that's necessary. Ibis doesn't have an API for this yet, and even if we did, there's no reason to copy the implementation over when we can let the error bubble up from Flink.

Reply:
No it's not just this. Among other things that Flink doesn't support the one that is breaking is that Flink doesn't have `GenerateSeries`, so if we were to call the original function the whole thing errors out.

Reply:
Also, `transforms.explode_to_unnest()` takes in a parameter `index_offset` and it doesn't look like `index_offset` can be `None`, which means that the compiled query always includes `WITH OFFSET` and will never execute on the Flink backend